src/clone.rs

```diff
@@ -0,0 +1,338 @@
+                ),
+            }));
+        }
+
+        let clone_manifest_store =
+            Arc::new(ManifestStore::new(&clone_path, Arc::clone(&object_store)));
+
+        let parent_manifest_store =
+            Arc::new(ManifestStore::new(&parent_path, Arc::clone(&object_store)));
+
+        let mut parent_manifest =
+            Self::load_initialized_manifest(Arc::clone(&parent_manifest_store)).await?;
+
+        let mut clone_manifest = Self::load_clone_manifest(
+            Arc::clone(&clone_manifest_store),
+            Arc::clone(&parent_manifest_store),
+            &mut parent_manifest,
+            &parent_path,
+            &parent_checkpoint,
+        )
+        .await?;
+
+        if !clone_manifest.db_state().initialized {
+            Self::copy_wal_ssts(
+                object_store.clone(),
+                clone_manifest.db_state(),
+                &parent_path,
+                &clone_path,
+            )
+            .await?;
+
+            let mut initialized_db_state = clone_manifest.db_state().clone();
+            initialized_db_state.initialized = true;
+            clone_manifest.update_db_state(initialized_db_state).await?;
+        }
+
+        Ok(())
+    }
+
+    async fn load_clone_manifest(
+        clone_manifest_store: Arc<ManifestStore>,
+        parent_manifest_store: Arc<ManifestStore>,
+        parent_manifest: &mut StoredManifest,
+        parent_path: &Path,
+        parent_checkpoint_id: &Option<Uuid>,
+    ) -> Result<StoredManifest, SlateDBError> {
+        if let Some(clone_manifest) = StoredManifest::try_load(clone_manifest_store.clone()).await?
+        {
+            Self::validate_existing_manifest(
+                &clone_manifest,
+                parent_path,
+                parent_manifest,
+                parent_checkpoint_id,
+            )?;
+            return Ok(clone_manifest);
+        }
+
+        let checkpoint_id = Uuid::new_v4();
+        let checkpoint_options = CheckpointOptions {
+            scope: Durable,
+            lifetime: None,
+            source: *parent_checkpoint_id,
+        };
+        let checkpoint = parent_manifest.new_checkpoint(checkpoint_id, &checkpoint_options)?;
+
+        let parent_db = manifest::DbLink {
+            path: parent_path.to_string(),
+            checkpoint_id,
+        };
+
+        let parent_checkpoint_manifest = if parent_manifest.id() == checkpoint.manifest_id {
+            parent_manifest.manifest().clone()
+        } else {
+            parent_manifest_store
+                .read_manifest(checkpoint.manifest_id)
+                .await?
+        };
+
+        let clone_manifest = StoredManifest::load_uninitialized_clone(
+            Arc::clone(&clone_manifest_store),
+            parent_db,
+            &parent_checkpoint_manifest,
+        )
+        .await?;
+
+        parent_manifest.write_checkpoint(checkpoint).await?;
+
+        Ok(clone_manifest)
+    }
+
+    fn validate_existing_manifest(
```

Comment:
this should also check that the parent manifest's destroyed_at field is not set. we haven't actually implemented that, but maybe add a todo and leave a note in https://github.com/slatedb/slatedb/issues/313 (or open another issue) to come back to it?

Reply:
Added a note on #313 .